src/main/java/io/jenkins/plugins/signpath/SignPathStepBase.java

```diff
@@ -122,15 +150,45 @@ protected String ensureNotNull(String input, String name) throws SignPathStepInv
         return input;
     }
 
-    protected String getWithGlobalConfig(String value, Function<SignPathPluginGlobalConfiguration, String> getter) {
-        if (value == null || value.isEmpty()) {
-            SignPathPluginGlobalConfiguration config = GlobalConfiguration.all().get(SignPathPluginGlobalConfiguration.class);
-            if (config != null) {
-                return getter.apply(config);
-            }
+    protected String getWithGlobalConfig(String stepLevelValue, Function<SignPathPluginGlobalConfiguration, String> globalValueGetter, String paramName) throws SignPathStepInvalidArgumentException {
+
+        SignPathPluginGlobalConfiguration config = GlobalConfiguration.all().get(SignPathPluginGlobalConfiguration.class);
+        if (config == null) {
+            throw new IllegalStateException("SignPathPluginGlobalConfiguration is not available. Ensure it is properly configured.");
+        }
+        String globalVal = globalValueGetter.apply(config);
+
+        if (globalVal == null || globalVal.isEmpty()) {
+            // global value is not set yet, fallback to the obsolete step-level value
+            return stepLevelValue;
         }
 
-        return value;
+        if (stepLevelValue == null || stepLevelValue.isEmpty()) {
+            // step level value is not set, use global value
+            return globalVal;
+        }
+
+        // here we have both values set. We enforce a rule that step level value should be identical to global value
+        if (!stepLevelValue.equals(globalVal)) {
+            throw new SignPathStepInvalidArgumentException(
+                String.format(
+                    "Parameter '%s' is configured globally to '%s' and cannot be changed in pipeline to '%s'.", 
```

Comment:
`Parameter <param-name> is configured globally to <global-value> and cannot be changed in pipeline to <step-value>` kind of implies that it's about the actual values rather than overriding a global parameter with a step value. Maybe rephrase?

Reply:
I specified the error message like that. I don't understand your phrase @Taha-cmd